pkg/snapshot/README.md

```diff
@@ -0,0 +1,203 @@
+
+![apis](./img/apis.png)
+
+2. Specify the Kubernetes resource names in Config Data
+
+By default, the created ApiDefinition resource will have the name `replace-me-{i}`. It is not the best name for a kubernetes resource and should be manually updated. You can store this meta-data in Config Data for each API first before exporting, so that you do not need to manually update the output file afterwards.
+
+![config-data](./img/config-data.png)
+
+## Usage
+```bash
+tyk-operator exports APIs and Security Policies from your Tyk Dashboard to Custom Resource that can be used with Tyk Operator
+
+Export API Definitions:
+   --snapshot <output_file>
+    	Pull a snapshot of ApiDefinitions from Tyk Dashboard and output as CR
+  --all
+    	Dump all APIs
+  --category <category_name>
+    	Dump APIs from specified category (default "operator")
+
+Export Security Policies:
+  --policy <output_file>
+    	Pull a snapshot of SecurityPolicies from Tyk Dashboard and output as CR
+```
+
+
+
+### Setting required environment variables
+
+Store the Tyk Dashboard connection parameters in environment variables before running `tyk-operator`, e.g.
+
+```bash
+TYK_MODE=pro TYK_URL=<TYK_URL> TYK_AUTH=<TYK_AUTH> TYK_ORG=<TYK_ORG> \
+./tyk-operator --snapshot <OUTPUT_FILE>
+```
+
+where
+- `<TYK_URL>`: Management URL of your Tyk Dashboard.
+- `<TYK_AUTH>`: Operator user API Key.
+- `<TYK_ORG>`: Operator user ORG ID.
+
+> For more details on how to obtain the URL and credentials, please visit [Tyk Docs](https://tyk.io/docs/tyk-stack/tyk-operator/installing-tyk-operator/#tyk-self-managed-hybrid).
+
+
+
+### Exporting API Definitions
+
+#### Specify Category to export
+
+By default, tyk-operator exports ApiDefinitions defined in `#operator` category. You can change default category via `--category` flag.
+
+```bash
+TYK_MODE=pro TYK_URL=<TYK_URL> TYK_AUTH=<TYK_AUTH> TYK_ORG=<TYK_ORG> ./tyk-operator --snapshot output.yaml --category k8s
+```
+
+You can also dump all ApiDefinitions from Dashboard via `--all` flag.
+
+```bash
+TYK_MODE=pro TYK_URL=<TYK_URL> TYK_AUTH=<TYK_AUTH> TYK_ORG=<TYK_ORG> ./tyk-operator --snapshot output.yaml --all
+```
+
+#### Output CR
+
+`tyk-operator` CLI creates an output file specified via `--snapshot` flag. Each ApiDefinition CR metadata has a default name  `replace-me-{i}` where `{i}` increases by each ApiDefinition.
+
+For example,
+```yaml
+apiVersion: tyk.tyk.io/v1alpha1
+kind: ApiDefinition
+metadata:
+  name: replace-me-0 # Default name for ApiDefinition CRs
+  namespace: default # Default namespace for ApiDefinition CRs
+spec:
+  ...
+```
+
+In order to specify CR metadata, you can use Config Data. For specified ApiDefinitions,
+snapshot CLI generates ApiDefinition CRs based on Config Data of that specific ApiDefinition.
+
+```json
+{
+  "k8sName": "metadata-name",
+  "k8sNs": "metadata-namespace"
```

Comment:
I would suggest that `k8sNs` should be renamed to `k8sNamespace`.
I would also recommend that if empty - it does not pick `default` but rather allows the user to choose when they kubectl apply.

e.g. `kubectl apply -f ./foo.yaml -n bar` in order to target the `bar` namespace.

Reply:
Thank you for the suggestion! It offers a better experience than it does presently.
I will update the PR accordingly

Reply:
@buraksekili Need to update screenshot. It is still referring to `k8sNs`.